cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -449,6 +462,35 @@ func (o *ScaleUpOrchestrator) ComputeExpansionOption(
 		return option
 	}
 
+	autoscalingOptions, err := nodeGroup.GetOptions(o.autoscalingContext.NodeGroupDefaults)
+	if err != nil {
+		klog.Errorf("Failed to get autoscaling options for node group %s: %v", nodeGroup.Id(), err)
+	}
+	if autoscalingOptions != nil && autoscalingOptions.AtomicScaleUp {
+		// If atomic, there should be no similar node groups for balancing
+		// Also the number of pods should be capped according to the node count.
+
+		// Limit the number of nodes we estimate to the max scale up possible.
+		currentTargetSize, err := nodeGroup.TargetSize()
+		if err != nil {
+			klog.Errorf("Failed to get node group size: %v", err)
+			return option
+		}
+		// When the node group has the atomic scale up option, the number of nodes
+		// that can be added to the node group should be capped according to the
+		// maximum size of the node group and not the 'MaxNodesPerScaleUp' option.
+		// Since we cannot change or update the limiter when we build the estimator
+		// with the 'autoscalingContext.EstimatorBuilder' we create a new estimator
+		// with a different limiter here.
+		nodeEstimationCap := nodeGroup.MaxSize() - currentTargetSize
+		limiter := estimator.NewThresholdBasedEstimationLimiter(nodeEstimationCap, 0)
+		estimator := estimator.NewBinpackingNodeEstimator(o.autoscalingContext.PredicateChecker, o.autoscalingContext.ClusterSnapshot, limiter, estimator.NewDecreasingPodOrderer())
+		option.NodeCount, option.Pods = estimator.Estimate(pods, nodeInfo, nodeGroup)
+		if option.NodeCount != nodeGroup.MaxSize() {
```

Comment:
Similar to one of the previous comments - I think this only works correctly as long as the node group can be only at 0 or MaxSize(). Is it guaranteed?

Reply:
In atomic scale up/down we should go from 0 -> max size -> 0. Node group should be either at 0 size or maximum size